src/pages/MovieInfo.jsx

```diff
@@ -0,0 +1,221 @@
+  z-index: 1;
+`;
+
+const BackButton = styled(Link)`
+  position: absolute;
+  top: 80px;
+  left: 40px;
+  display: flex;
+  align-items: center;
+  text-decoration: none;
+  color: #fff;
+  font-size: 16px;
+  font-weight: bold;
+
+  img {
+    margin-right: 5px;
+    width: 24px;
+    height: 24px;
+  }
+
+  &:hover {
+    color: #ccc;
+  }
+`;
+
+const ContentContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: flex-start;
+  justify-content: center;
+  margin-top: -600px; /* Overlap content with backdrop */
+  gap: 5px;
+  position: relative; /* Keeps the content over the backdrop */
+
+  @media (max-width: 768px) {
+    flex-direction: column;
+    align-items: center;
+    margin-top: -900px;
+    padding: 0 20px;
+  }
+`;
+
+const PosterContainer = styled.div`
+  flex-shrink: 0;
+  position: relative;
+
+  .movie-poster {
+    width: 300px;
+    border: solid white; 
+    position: relative; /* Keeps it positioned properly within its container */
+    z-index: 2; /* Ensures poster is above backdrop */
+
+    @media (min-width: 769px) {
+      width: 300px;
+      margin-left: 20px;
+      margin-top: -250px;
+    }
+
+    @media (max-width: 768px) {
+      width: 250px;
+    }
+  }
+`;
+
+const DetailsContainer = styled.div`
+  flex-grow: 1;
+  z-index: 2;
+  max-width: 600px;
+  padding: 20px;
+  color: #fff;
+
+  @media (max-width: 768px) {
+    text-align: left;
+  }
+
+  @media (min-width: 769px) {
+      margin-top: -40px;
+    }
+`;
+
+
+const RatingContainer = styled.div`
+  display: inline-block;
+  align-items: center;
+  background-color: white;
+  padding: 0 5px;
+  width: fit-content;
+  font-size: 20px;
+  font-weight: bold;
+  margin: 5px 0px;
+
+  span.star {
+    color: gold;
+    margin-right: 8px;
+    font-size: 24px; /* Adjust the size of the star */
+  }
+
+  span.rating {
+    color: black;
+  }
+`;
+
+export const MovieInfo = () => {
+  const { id } = useParams();
+  const [movie, setMovie] = useState(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const apiEnv = import.meta.env.VITE_TMDB_KEY;
+    const fetchMovieDetails = async () => {
+      try {
+        const response = await fetch(
+          `https://api.themoviedb.org/3/movie/${id}?api_key=${apiEnv}`
+        );
+        const data = await response.json();
+        setMovie(data);
+      } catch (error) {
+        console.error("Error fetching movie details:", error);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchMovieDetails();
+  }, [id]);
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!movie) {
+    return <div>Movie not found</div>;
+  }
+
+  return (
+    <MovieInfoWrapper>
+      <BackdropContainer>
+        <BackdropImage
+          src={`https://image.tmdb.org/t/p/w1280${movie.backdrop_path}`}
+          alt={movie.title}
+          className="movie-backdrop"
+        />
+      </BackdropContainer>
+      <BackButton to="/">
```

Comment:
It would be nice if this link could be a little bit "smarter". So that if I'm on the Upcoming page and I click on a movie, then the link would take me back to the Upcoming page. Now it always leads to the home page.

Reply:
I know, I have tried that but then the back button to the now playing movies/home page won´t work, I think it has to do with that I have placed the now playing movies on the home page and not as a separate route. I will look into this more when I have time.